pkg/apis/navigator/v1alpha1/types.go

```diff
@@ -31,8 +31,14 @@ type CassandraClusterSpec struct {
 	NavigatorClusterConfig `json:",inline"`
 
 	NodePools []CassandraClusterNodePool `json:"nodePools"`
-	Image     ImageSpec                  `json:"image"`
-	CqlPort   int32                      `json:"cqlPort"`
+
+	// Image describes the database image to use
+	Image *ImageSpec `json:"image"`
```

Comment:
should this now be marked +optional?

Reply:
I can't tell if it's necessary. 
`ElasticsearchClusterSpec.Image` hasn't got `+optional`, nor `omitempty`.

The `+optional` comment is used here:

```
fgrep -B1 -A10 '+optional' vendor/k8s.io/kube-openapi/pkg/generators/openapi.go

// hasOptionalTag returns true if the member has +optional in its comments or
// omitempty in its json tags.
func hasOptionalTag(m *types.Member) bool {
	hasOptionalCommentTag := types.ExtractCommentTags(
		"+", m.CommentLines)[tagOptional] != nil
	hasOptionalJsonTag := strings.Contains(
		reflect.StructTag(m.Tags).Get("json"), "omitempty")
	return hasOptionalCommentTag || hasOptionalJsonTag
}

type identityNamer struct{}

```

I'll leave it for now, but opened https://github.com/jetstack/navigator/issues/265 to discuss further.